visualization/src/point_picking_event.cpp

```diff
@@ -203,52 +205,51 @@ pcl::visualization::PointPickingCallback::performAreaPick (vtkRenderWindowIntera
     {
       vtkActor* actor = vtkActor::SafeDownCast(prop);
-      if(actor)
-      {
-        actors_->AddItem(actor);
-
-        std::vector<int> actorIndices;
-        vtkPolyData* pd = vtkPolyData::SafeDownCast(actor->GetMapper()->GetInput());
-
-        if(pd->GetPointData()->HasArray("Indices"))
+      if (!actor)
+        { continue; }
+        
+      vtkPolyData* pd = vtkPolyData::SafeDownCast(actor->GetMapper()->GetInput());
+      if(pd->GetPointData()->HasArray("Indices"))
             pd->GetPointData()->RemoveArray("Indices");
 
+      vtkSmartPointer<vtkIdTypeArray> IDs = vtkSmartPointer<vtkIdTypeArray>::New();
+      IDs->SetNumberOfComponents(1);
+      IDs->SetName("Indices");
+      for(vtkIdType i = 0; i < pd->GetNumberOfPoints(); i++)
+        IDs->InsertNextValue(i);
 
-        vtkSmartPointer<vtkIdTypeArray> IDs = vtkSmartPointer<vtkIdTypeArray>::New();
-        IDs->SetNumberOfComponents(1);
-        IDs->SetName("Indices");
-        for(vtkIdType i = 0; i < pd->GetNumberOfPoints(); i++)
-          IDs->InsertNextValue(i);
+      pd->GetPointData()->AddArray(IDs);
 
-        pd->GetPointData()->AddArray(IDs);
+      vtkPlanes* frustum = picker->GetFrustum();
 
-        vtkPlanes* frustum = picker->GetFrustum();
-
-        vtkSmartPointer<vtkExtractGeometry> extract_geometry = vtkSmartPointer<vtkExtractGeometry>::New();
-        extract_geometry->SetImplicitFunction(frustum);
+      vtkSmartPointer<vtkExtractGeometry> extract_geometry = vtkSmartPointer<vtkExtractGeometry>::New();
+      extract_geometry->SetImplicitFunction(frustum);
   #if VTK_MAJOR_VERSION < 6
-        extract_geometry->SetInput(pd);
+      extract_geometry->SetInput(pd);
   #else
-        extract_geometry->SetInputData(pd);
+      extract_geometry->SetInputData(pd);
   #endif
 
-        extract_geometry->Update();
+      extract_geometry->Update();
 
-        vtkSmartPointer<vtkVertexGlyphFilter> glyph_filter = vtkSmartPointer<vtkVertexGlyphFilter>::New ();
-        glyph_filter->SetInputConnection (extract_geometry->GetOutputPort ());
-        glyph_filter->Update ();
+      vtkSmartPointer<vtkVertexGlyphFilter> glyph_filter = vtkSmartPointer<vtkVertexGlyphFilter>::New ();
+      glyph_filter->SetInputConnection (extract_geometry->GetOutputPort ());
+      glyph_filter->Update ();
 
-        vtkPolyData* selected = glyph_filter->GetOutput ();
+      vtkPolyData* selected = glyph_filter->GetOutput ();
 
-        vtkIdTypeArray* GlobalIDs  = vtkIdTypeArray::SafeDownCast(selected->GetPointData()->GetArray("Indices"));
-        for (vtkIdType i = 0; i < selected->GetNumberOfPoints (); i++)
-        {
-          actorIndices.push_back(static_cast<int>(GlobalIDs->GetValue(i)));
-          pt_numb++;
-        }
-        indices.push_back(actorIndices);
-      }
+      vtkIdTypeArray* GlobalIDs  = vtkIdTypeArray::SafeDownCast(selected->GetPointData()->GetArray("Indices"));
+      actors->AddItem(actor);
```

Comment:
Move this line up at the top or down at the bottom (preferred)

Reply:
agree